rustls/src/error.rs

```diff
@@ -94,6 +94,15 @@ pub enum Error {
     /// or too large.
     BadMaxFragmentSize,
 
+    /// This [`rustls::crypto::signer::SigningKey`] has not implemented [`rustls::crypto::signer::SigningKey::public_key()`].
+    UnimplementedSubjectPublicKeyInfo,
+
+    /// We failed to parse DER into an end-entity certificate.
+    FailedToParseEndEntityCert(webpki::Error),
+
+    /// We could not find a key's SubjectPublicKeyInfo.
+    KeyDoesNotContainSubjectPublicKeyInfo,
```

Comment:
Not completely sure of the distinction between this and `UnimplementedSubjectPublicKeyInfo`?

Reply:
It feels like `Error::InvalidCertificate` would be appropriate for this case, the SPKI is mandatory in the `TBSCertificate` per RFC 5280. It's absence indicates an invalid cert.

Reply:
> It feels like Error::InvalidCertificate would be appropriate for this case, the SPKI is mandatory in the TBSCertificate per RFC 5280. It's absence indicates an invalid cert.

This makes sense. I removed the error and replaced the code that necessitated it with `Error::InvalidCertificate(CertificateError::BadEncoding)`, which aligns more with this perspective